src/tree/localProject/LocalProjectTreeItem.ts

```diff
@@ -62,14 +63,27 @@ export class LocalProjectTreeItem extends LocalProjectTreeItemBase implements Di
         this._localFunctionsTreeItem = new LocalFunctionsTreeItem(this);
     }
 
-    public async getHostUrl(context: IActionContext): Promise<string> {
+    public async getHostRequest(context: IActionContext): Promise<FuncHostRequest> {
         let port = runningFuncPortMap.get(this.workspaceFolder);
         if (!port) {
             const funcTask: Task | undefined = (await tasks.fetchTasks()).find(t => t.scope === this.workspaceFolder && isFuncHostTask(t));
             port = await getFuncPortFromTaskOrProject(context, funcTask, this.effectiveProjectPath);
         }
 
-        return `http://localhost:${port}`;
+        const url = `http://localhost:${port}`;
+        try {
+            await requestUtils.sendRequestWithExtTimeout(context, { url: url, method: 'GET' });
```

Comment:
If the http localhost doesn't exist, this just throws an error immediately, right?

Reply:
Yeah it errors out quickly in my experience